docs/sql/data_types/union.md

```diff
@@ -0,0 +1,86 @@
+
+`UNION` types must always have at least one member, and while they can contain multiple members of the same type, the tag names must be unique. `UNION` types can have at most 256 members.
+ 
+Under the hood, `UNION` types are implemented on top of `STRUCT` types, and simply keep the "tag" as the first entry.
+
+`UNION` values can be created with the [`UNION_VALUE(tag := expr)`](../functions/nested#union-functions) function or by [casting from a member type](#casting-to-unions).
+
+### Example 
+```sql
+-- Create a table with a union column
+CREATE TABLE tbl1(u UNION(num INT, str VARCHAR));
+
+-- Any type can be implicitly cast to a union containing the type.
+-- Any union can also be implicitly cast to a another union if 
+-- the source union members are a subset of the targets.
+-- Note: only if the cast is unambiguous! 
+-- More details in the 'Union casts' section below.
+INSERT INTO tbl1 values (1) , ('two') , (union_value(str := 'three'));
+-- Union use the member types varchar cast functions when casting to varchar.
+SELECT u from tbl1;
+-- returns:
+--    1
+--    two
+--    three
+-- Select all the 'str' members
+SELECT union_extract(u, 'str') FROM tbl1;
+-- Alternatively, you can use 'dot syntax' like with structs
+SELECT u.str FROM tbl1;
+-- returns: 
+--    NULL
+--    two
+--    three
+
+-- Select the currently active tag from the union as an enum.
+SELECT union_tag(u) FROM tbl1;
+-- returns:
+--    num
+--    str
+--    str
+```
+
+## Union casts
+Compared to other nested types, `UNION`s allow a set of implicit casts to facilitate unintrusive and natural usage when working with their members as "subtypes".
+However, these casts have been designed with two principles in mind, to avoid ambiguity and to avoid casts that could lead to loss of information. This prevents `UNION`s from being completely "transparent", while still allowing `UNION` types to have a "supertype" relationship with their members.
+
+Thus `UNION` types can't be implicitly cast to any of their member types in general, since the information in the other members not matching the target type would be "lost". If you want to coerce a `UNION` into one of its members, you should use the `union_extract` function explicitly instead.
+
+The only exception to this is when casting a `UNION` to `VARCHAR`, in which case the members will all use their corresponding `VARCHAR` casts. Since everything can be cast to `VARCHAR`, this is "safe" in a sense. 
+
+### Casting to unions
+A type can always be implicitly cast to a `UNION` if it can be implicitly cast to one of the `UNION` member types.
+
+- If there are multiple candidates, the lowest cast cost is used. 
```

Comment:
What is this cost you're referring to?

Reply:
Ah, this might not be documented elsewhere, but all implicit casts carry a "cost" to disambiguate when resolving types. e.g casting a float to a int is cheaper than casting a float to a varchar. I could world this differently here and just say "the candidate is selected based on the normal casting priorities" and try to avoid exposing the "cost" implementation detail.